### PR TITLE
defaultImageRepository instead of imageOverrideRepository

### PR DESCRIPTION
**Pull request by ShubhamRwt**
### Type of change

_Select the type of your PR_

- Enhancement / new feature
- Refactoring

### Description

This PR fixes the issue mentioned in #4014.
Earlier we were making use of imageOverrideTag, imageOverrideRepository and imageOverrideRegistry. But the problem with this approach was if anyone set them then it will override the tags, registry and repository for all the images ie.all the images will now use same tags, registry, and repository. If someone want to set the registry, repository or tag for a single image he can't do that since all the override will apply to all the image. This PR resolves this issue by introducing defaultImageRepository, defaultImageRegistry, defaultImageTag which points to default repo, registry and tags. If the user wants wants to set these parameters for a single image now they can do it easily.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: packaging/helm-charts/helm3/strimzi-kafka-operator/values.yaml [hunk 1]
```diff
@@ -32,94 +36,94 @@ securityContext: {}
...
-    registry: quay.io
-    repository: strimzi
+    registry: ""
+    repository: ""
     name: kafka
-    tagPrefix: latest
+    tagPrefix: ""
 kanikoExecutor:
   image:
     registry: quay.io
```
Comment (scholzj) [inline packaging/helm-charts/helm3/strimzi-kafka-operator/values.yaml:123]: Should this still be here?

**Review by scholzj — COMMENTED** [review]
One nit. Also, I think this deserves mentioning in the CHANGELOG _Changes, deprecations and removals_ section (https://github.com/strimzi/strimzi-kafka-operator/blob/main/CHANGELOG.md#changes-deprecations-and-removals)

**Review by scholzj — COMMENTED** [review]
I think you will also need to update the release tooling which was updating the `tag` / `tagPrefix` fields and now will need to update `defaultImageTag`. I think you also need to update the README file (including the related release tooling).

**Review by scholzj — APPROVED** [review]

**Review by sknot-rh — APPROVED** [review]
This will cause conflicts in https://github.com/strimzi/strimzi-kafka-operator/pull/5407
I am happy with fixing them, so this can be merged.

**Pull request merged**